src/tiledb/cloud/dag/dag.py

```diff
@@ -1801,7 +1862,56 @@ def _update_status(self) -> None:
                     futures.execute_callbacks(nd, cbs)
                 with self._lifecycle_condition:
                     self._set_status(Status.FAILED)
-                raise  # Bail out and fail loudly.
+                raise  # Bail out and fail loudly
+
+    def _tdb_to_json(self, override_name: Optional[str] = None) -> Dict[str, Any]:
+        """Converts this DAG to a registerable format.
+
+        :param override_name: Name to override DAG conversion.
+        :return: Mapping of DAG tree to json for submission to REST.
+        """
+
+        nodes = _topo_sort_nodes(self.nodes)
+
+        existing_names = set(self.nodes_by_name.keys())
+
+        node_jsons = [n.to_registration_json(existing_names) for n in nodes]
+
+        for n, n_json in zip(nodes, node_jsons):
+            n_json["depends_on"] = [str(parent.id) for _, parent in n.parents.items()]
+
+        return {
+            "name": override_name or self.name,
+            "nodes": node_jsons,
+        }
+
+    def register(
+        self,
+        override_name: Optional[str] = None,
+    ) -> str:
+        """Register DAG to TileDB.
+
+        :param override_name: Name to register DAG as. Uses self.name as default.
+        :return: Registered name of task graph.
+        """
+
+        tg_name = override_name or self.name
+
+        if not tg_name:
+            raise ValueError(
+                "Must specify registration name to DAG.name or override_name."
+            )
+
+        registration.register(
+            graph=self,
+            name=tg_name,
+            namespace=self.namespace,
```

Comment:
At first, I was wondering where all the new private methods were coming from, and then I realized you had copied them from the taskgraph module, and that they are called by the JSON serializer. I think this is an okay way to go. Could you look into another approach, to make sure that we're not missing out on a chance to do this with less code?

It occurs to me that `dag.compute()` uses our task graph builder to produce JSON that is sent to the server to be executed. Isn't this the same JSON that we would use when registering? Could your new `register()` method run `_build_batch_taskgraph()`, for example, to produce the JSON?

If this is a dumb or not workable idea, I will gladly drop it.

Reply:
@spencerseale I have a better idea, one that will make new modules and new interfaces unnecessary for now. Instead of importing `tiledb.cloud.taskgraphs.registration` (which tangles our modules together) and calling `registration.register()`, copy the body of `registration.register()` into `Dag.register()`.

A little bit of code duplication, it's only two copies, doesn't bother me at all.